examples/i18n/src/resource.rs

```diff
@@ -0,0 +1,33 @@
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct Resource(HashMap<String, String>);
```

Comment:
My gut feeling tells me we don't need this struct at all. Try to eliminate it as the last step. If it's not possible I'll write more comments for it.

Reply:
We could name the FTL files like the Language Identifier and just load the FTL string when language is set or changed. This would eliminate module resource.rs completely. I'll try that ...

Reply:
If I understand correctly: include_str! expects a string literal so we can't build the path to the file dynamically.
I found [load_file](https://crates.io/crates/load_file) which allows that. For me it looks like that this supports requirement 5 "be able to provide translations by muggles.".

Ok, I head into this direction ...

Reply:
@MartinKavik Looks like load_file loads files at runtime but in the actual state of I18n I have to compile the translation into wasm. I thought about adding the FTL strings to every Lang variant or to I18n. But Lang and I18n just handle one language at a time while I would have to compile all FTL strings and keep track of which FTL string belongs to which language identifier.
Not sure how to handle that without complicating the example unnecessarily.

Reply:
I need to dive deeper and experiment a bit. I'll look at it tomorrow and probably create a PR to your branch. So please finish what you have.

Do we want lazy-load files with FTL strings?

Reply:
Thank you.

Nothing to commit. Working tree clean.

Regarding the delivery of the FTL strings: I think it depends. In case of a small web app supporting just a few languages it would be okay to compile everything into the wasm binary and load it all upfront. As soon as the web app grows or a lot of different languages have to be supported it would be nice to compile everything into wasm but load only the needed language. Is that possible in wasm?
One step further for a more dynamic multilingual approach where we can change the set of translations during runtime I think the web app will load the FTLs from the web server. I wouldn't implement this in this example. I think the focus should be to provide an example easy to adopt even for beginners. Having said that loading FTLs dynamically from a web server could be an additional example.